File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.java
    import java.util.*;
    import java.io.*;
    public class Main {

    	public static void main(String[] args) throws IOException {
    	BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
    	BufferedWriter bw=new BufferedWriter(new OutputStreamWriter(System.out));
    	int num=Integer.parseInt(br.readLine());
    	int arr[]=new int[11];
    	arr[0]=1;
    	arr[1]=2;
    	arr[2]=4;
    	for(int i=0;i<num;i++) {
    		int n=Integer.parseInt(br.readLine());
    		for(int j=3;j<n;j++) {
    			arr[j]=arr[j-1]+arr[j-2]+arr[j-3];
    		}
    		System.out.println(arr[n-1]);
    	}
}
    	}
    

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py
num = int(input())
arr = [0]*11
arr[0] = 1
arr[1] = 2
arr[2] = 4

for i in range(num):
    n = int(input())
    for j in range(3,n):
        arr[j] = arr[j-1]+arr[j-2]+arr[j-3]
    print(arr[n-1])
